rollout/controller.go

```diff
@@ -419,6 +422,14 @@ func (c *Controller) writeBackToInformer(ro *v1alpha1.Rollout) {
 		return
 	}
 	un := unstructured.Unstructured{Object: obj}
+	gvk := un.GetObjectKind().GroupVersionKind()
+	if len(gvk.Version) == 0 || len(gvk.Group) == 0 || len(gvk.Kind) == 0 {
+		un.GetObjectKind().SetGroupVersionKind(schema.GroupVersionKind{
+			Group:   v1alpha1.SchemeGroupVersion.Group,
+			Kind:    rollouts.RolloutKind,
+			Version: v1alpha1.SchemeGroupVersion.Version,
+		})
+	}
```

Comment:
Could you please add a comment explaining the reason for doing this? I still consider this a workaround that we could hopefully remove one day. Also, note that this only resolves the issue with Rollout kind but not any others (AnalysisRun, Experiment)